### PR TITLE
#2410 splitting and renaming DynStencils to LFRicStencils

### LINKED ISSUE #2410: Splitting and renaming DynStencils from dynamo0p3.py
sub-issue of #2235

### PR DESCRIPTION
**Pull request by mo-lottieturner**
closes #2410

### CODE REVIEW

**Comment by mo-lottieturner** [comment]
have tried moving each file's LFRicStencils import to the top level individually, but unfortunately none of them worked

**Comment by TeranIvy** [comment]
Taking this for review.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+        self._unique_direction_args = []
+        direction_names = []
+        for call in self._calls:
+            for idx, arg in enumerate(call.arguments.args):
+                if arg.stencil and arg.stencil.direction_arg:
+                    if arg.stencil.direction_arg.is_literal():
+                        raise GenerationError(
+                            f"Kernel {call.name}, metadata arg {idx}, a "
+                            f"literal is not a valid value for a stencil "
+                            f"direction")
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:104]: I would suggest adding a full stop in this error message. The related test will need to be adjusted, too.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+                    if arg.stencil.direction_arg.text.lower() not in \
+                       ["x_direction", "y_direction"]:
+                        if arg.stencil.direction_arg.text not in \
+                           direction_names:
+                            direction_names.append(
+                                arg.stencil.direction_arg.text)
+                            self._unique_direction_args.append(arg)
+
+        # list of stencil args with an extent variable passed in. The same
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:113]: Upper case "List", please.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+from psyclone.f2pygen import (AssignGen, CommentGen, DeclGen,
+                              IfThenGen, TypeDeclGen, UseGen)
+from psyclone.psyir.symbols import ScalarType
+
+class LFRicStencils(LFRicCollection):
+    '''
+    Stencil information and code generation associated with a PSy-layer
+    routine or Kernel stub.
+
+    :param node: the Invoke or Kernel stub for which to provide stencil info.
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:59]: Please remove `\` in descriptions here and other lines in the file where applicable.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 4]
```diff
@@ -0,0 +1,649 @@
...
+        :param arg: kernel argument with which stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param str context: a context for this stencil (e.g. "size" or \
+                            "direction").
+
+        :returns: unique string identifying the stencil for this argument.
+        :rtype: str
+
+        :raises GenerationError: if an explicit stencil extent is found in \
+                                 the meta-data for the kernel argument.
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:157]: "metadata" here and throughout the file, I think. Please also remove `\` from descriptions and add a blank line above the triple quotes as said above.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 5]
```diff
@@ -0,0 +1,649 @@
...
+    def map_name(self, arg):
+        '''
+        Creates and registers a name for the stencil map associated with the
+        supplied kernel argument.
+
+        :param arg: kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+
+        :returns: a valid unique map name for a stencil in the PSy layer.
+        :rtype: str
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:180]: Please add a blank line above the triple quotes.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 6]
```diff
@@ -0,0 +1,649 @@
...
+        dofmap array dimensions.
+
+        :param symtab: symbol table that will contain (or already contains) \
+            the symbol with this name.
+        :type symtab: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param arg: the kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+
+        :returns: a Fortran variable name for the max stencil branch length.
+        :rtype: str
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:253]: Please add a blank line above the triple quotes.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 7]
```diff
@@ -0,0 +1,649 @@
...
+        unique = LFRicStencils.stencil_unique_str(arg, "map")
+        return self._symbol_table.find_or_create_tag(unique, root_name).name
+
+    @staticmethod
+    def dofmap_symbol(symtab, arg):
+        '''
+        Creates and registers a symbol for the stencil dofmap associated with
+        the supplied kernel argument.
+
+        :param symtab: symbol table that will contain (or already contains) \
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:192]: Please remove the `\` character and align "the symbol" in the line below with "symbol table".

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 8]
```diff
@@ -0,0 +1,649 @@
...
+                                           ScalarType.Intrinsic.INTEGER,
+                                           tag=unique)
+
+    @staticmethod
+    def dofmap_size_symbol(symtab, arg):
+        '''
+        Create a valid symbol for the size (in cells) of a stencil
+        dofmap in the PSy layer.
+
+        :param symtab: symbol table that will contain (or already contains) \
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:218]: Please remove the `\` character and align "the symbol" in the line below with "symbol table".

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 9]
```diff
@@ -0,0 +1,649 @@
...
+
+    @staticmethod
+    def max_branch_length_name(symtab, arg):
+        '''
+        Create a valid unique name for the maximum length of a stencil branch
+        (in cells) of a 2D stencil dofmap in the PSy layer. This is required
+        in the kernels for defining the maximum possible length of one of the
+        dofmap array dimensions.
+
+        :param symtab: symbol table that will contain (or already contains) \
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:246]: Please remove the `\` character and align "the symbol" in the line below with "symbol table".

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 10]
```diff
@@ -0,0 +1,649 @@
...
+        :rtype: str
+        '''
+        root_name = arg.name + "_max_branch_length"
+        unique = LFRicStencils.stencil_unique_str(arg, "length")
+        return symtab.find_or_create_integer_symbol(root_name, tag=unique).name
+
+    def _unique_max_branch_length_vars(self):
+        '''
+        :returns: list of all the unique max stencil extent argument names in
+                  this kernel call for cross2d stencils.
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:262]: I would suggest upper case "CROSS2D". Please also insert a blank line below the `rtype` description.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 11]
```diff
@@ -0,0 +1,649 @@
...
+
+        return names
+
+    def _declare_unique_max_branch_length_vars(self, parent):
+        '''
+        Declare all unique max branch length arguments as integers with intent
+        in and add the declaration as a child of the parent argument passed
+        in.
+
+        :param parent: the node in the f2pygen AST to which to add the \
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:278]: Please remove the `\` character from decsriptions..

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 12]
```diff
@@ -0,0 +1,649 @@
...
+                entity_decls=self._unique_max_branch_length_vars(), intent="in"
+            ))
+
+    @staticmethod
+    def direction_name(symtab, arg):
+        '''
+        Creates a Fortran variable name to hold the direction of the stencil
+        associated with the supplied kernel argument.
+
+        :param symtab: symbol table that will contain (or already contains) \
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:298]: Please remove the `\` character and align "the symbol" in the line below with "symbol table".

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 13]
```diff
@@ -0,0 +1,649 @@
...
+            the symbol with this name.
+        :type symtab: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param arg: the kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+
+        :returns: a Fortran variable name for the stencil direction.
+        :rtype: str
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:305]: Please insert a blank line below this one.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 14]
```diff
@@ -0,0 +1,649 @@
...
+        '''
+        root_name = arg.name+"_direction"
+        unique = LFRicStencils.stencil_unique_str(arg, "direction")
+        return symtab.find_or_create_integer_symbol(root_name, tag=unique).name
+
+    @property
+    def _unique_extent_vars(self):
+        '''
+        :returns: list of all the unique extent argument names in this \
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:314]: Please remove the `\` character from descriptions.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 15]
```diff
@@ -0,0 +1,649 @@
...
+                  invoke or kernel call.
+        :rtype: list of str
+
+        :raises InternalError: if neither self._kernel or self._invoke are set.
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:318]: I would suggest single quotations (') around `'self._kernel'` and `'self._invoke'`.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 16]
```diff
@@ -0,0 +1,649 @@
...
+
+        '''
+        if self._invoke:
+            names = [arg.stencil.extent_arg.varname for arg in
+                     self._unique_extent_args]
+        elif self._kernel:
+            names = [self.dofmap_size_symbol(self._symbol_table, arg).name
+                     for arg in self._unique_extent_args]
+        else:
+            raise InternalError("_unique_extent_vars: have neither Invoke "
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:328]: I would suggest specifying the name of the parent class in this error message to make it easier. That is "LFRicStencils._unique_extent_vars: have ...". The relevant test will need to be adjusted.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 17]
```diff
@@ -0,0 +1,649 @@
...
+                                "or Kernel. Should be impossible.")
+        return names
+
+    def _declare_unique_extent_vars(self, parent):
+        '''
+        Declare all unique extent arguments as integers with intent in and
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:334]: I would suggest single quotations around `'in'` here to mark it is code.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 18]
```diff
@@ -0,0 +1,649 @@
...
+        for arg in self._kern_args:
+            if arg.descriptor.stencil['type'] == "cross2d":
+                names.append(arg.name + "_max_branch_length")
+
+        return names
+
+    def _declare_unique_max_branch_length_vars(self, parent):
+        '''
+        Declare all unique max branch length arguments as integers with intent
+        in and add the declaration as a child of the parent argument passed
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:275]: I would suggest single quotations around `'in'` here to mark it is code (intent).

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 19]
```diff
@@ -0,0 +1,649 @@
...
+                                "or Kernel. Should be impossible.")
+        return names
+
+    def _declare_unique_extent_vars(self, parent):
+        '''
+        Declare all unique extent arguments as integers with intent in and
+        add the declaration as a child of the parent argument passed
+        in.
+
+        :param parent: the node in the f2pygen AST to which to add the \
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:338]: Please remove the `\` character from the description (and others in the file).

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 20]
```diff
@@ -0,0 +1,649 @@
...
+                    entity_decls=self._unique_extent_vars, intent="in"
+                ))
+
+    @property
+    def _unique_direction_vars(self):
+        '''
+        :returns: a list of all the unique direction argument names in this \
+                  invoke call.
+        :rtype: list of str
+        '''
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:374]: Please insert a blank line above '''.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 21]
```diff
@@ -0,0 +1,649 @@
...
+        for arg in self._unique_direction_args:
+            if arg.stencil.direction_arg.varname:
+                names.append(arg.stencil.direction_arg.varname)
+            else:
+                names.append(arg.name+"_direction")
+        return names
+
+    def _declare_unique_direction_vars(self, parent):
+        '''
+        Declare all unique direction arguments as integers with intent in
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:385]: I would suggest single quotations around `'in'` here to mark it is code.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 22]
```diff
@@ -0,0 +1,649 @@
...
+                               entity_decls=self._unique_direction_vars,
+                               intent="in"))
+
+    @property
+    def unique_alg_vars(self):
+        '''
+        :returns: list of the names of the extent and direction arguments \
+                  supplied to the PSy routine from the Algorithm layer.
+        :rtype: list of str
+        '''
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:408]: Please insert a blank line above ‘’’.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 23]
```diff
@@ -0,0 +1,649 @@
...
+        :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
+
+        '''
+        self._declare_unique_extent_vars(parent)
+        self._declare_unique_direction_vars(parent)
+        self._declare_maps_invoke(parent)
+
+    def _stub_declarations(self, parent):
+        '''
+        Declare all stencil-related quanitites for a Kernel stub.
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:426]: I would suggest "Declares".

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 24]
```diff
@@ -0,0 +1,649 @@
...
+    def initialise(self, parent):
+        '''
+        Adds in the code to initialise stencil dofmaps to the PSy layer.
+
+        :param parent: the node in the f2pygen AST to which to add the \
+                       initialisations.
+        :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
+
+        :raises GenerationError: if an unsupported stencil type is encountered.
+        '''
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:446]: Please insert a blank line above ‘’’ (please also do that through the module where missing).

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 25]
```diff
@@ -0,0 +1,649 @@
...
+                    parent, datatype="integer",
+                    kind=api_config.default_kind["integer"], intent="in",
+                    dimension=",".join([arg.function_space.ndf_name,
+                                        dofmap_size_name]),
+                    entity_decls=[self.dofmap_symbol(symtab, arg).name]))
+
+
+# ---------- Documentation utils -------------------------------------------- #
+# The list of module members that we wish AutoAPI to generate
+# documentation for. (See https://psyclone-ref.readthedocs.io)
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:648]: I would suggest "documentation for (see https://psyclone-ref.readthedocs.io)."

**Review by TeranIvy — CHANGES_REQUESTED** [review]
Thanks for the refactoring! I have some minor style remarks and a request to update the bransh with the master (see inline comments). I grepped the code and documentation for the old class name and all is fine. Also, the tests with compilation pass.

File: src/psyclone/domain/lfric/kern_call_acc_arg_list.py [hunk 26]
```diff
@@ -109,8 +109,8 @@ def stencil(self, arg, var_accesses=None):
         '''
         # Import here to avoid circular dependency
         # pylint: disable=import-outside-toplevel
-        from psyclone.dynamo0p3 import DynStencils
-        var_name = DynStencils.dofmap_symbol(self._kern.root.symbol_table,
+        from psyclone.domain.lfric.lfric_stencils import LFRicStencils
```
[the remark this answers is not in the document]
Comment (mo-lottieturner) [reply src/psyclone/domain/lfric/kern_call_acc_arg_list.py:112]: these are both due to the Dyn to LFRic change in the line above - fixed [ed1ea8b](https://github.com/stfc/PSyclone/pull/2423/commits/ed1ea8bf5fb9f5d179e129cb62283631656f1e7e)

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 27]
```diff
@@ -0,0 +1,649 @@
...
+
+from psyclone.configuration import Config
+from psyclone.domain.lfric.lfric_collection import LFRicCollection
+from psyclone.domain.lfric.lfric_constants import LFRicConstants
+from psyclone.errors import GenerationError, InternalError
+from psyclone.f2pygen import (AssignGen, CommentGen, DeclGen,
+                              IfThenGen, TypeDeclGen, UseGen)
+from psyclone.psyir.symbols import ScalarType
+
+class LFRicStencils(LFRicCollection):
```
[the remark this answers is not in the document]
Comment (mo-lottieturner) [reply src/psyclone/domain/lfric/lfric_stencils.py:55]: [ed1ea8b](https://github.com/stfc/PSyclone/pull/2423/commits/ed1ea8bf5fb9f5d179e129cb62283631656f1e7e)

File: src/psyclone/tests/domain/lfric/lfric_stencil_test.py [hunk 28]
```diff
@@ -1976,14 +1976,14 @@ def test_dynstencils_extent_vars_err(monkeypatch):
 
 
 def test_dynstencils_err():
-    ''' Check that DynStencils.initialise and DynStencils._declare_maps_invoke
+    ''' Check that LFRicStencils.initialise and LFRicStencils._declare_maps_invoke
```
[the remark this answers is not in the document]
Comment (mo-lottieturner) [reply src/psyclone/tests/domain/lfric/lfric_stencil_test.py]: [ed1ea8b](https://github.com/stfc/PSyclone/pull/2423/commits/ed1ea8bf5fb9f5d179e129cb62283631656f1e7e)

File: src/psyclone/domain/lfric/lfric_kern.py [hunk 29]
```diff
@@ -609,16 +609,15 @@ def gen_stub(self):
...
         # pylint: disable=import-outside-toplevel
-        from psyclone.domain.lfric import LFRicScalarArgs, LFRicFields
+        from psyclone.domain.lfric import LFRicFields, LFRicScalarArgs, LFRicStencils
         from psyclone.dynamo0p3 import (DynCellIterators, DynDofmaps,
                                         DynFunctionSpaces, DynCMAOperators,
                                         DynBoundaryConditions,
                                         DynLMAOperators, LFRicMeshProperties,
-                                        DynBasisFunctions, DynStencils,
-                                        DynReferenceElement)
+                                        DynBasisFunctions, DynReferenceElement)
```
[the remark this answers is not in the document]
Comment (mo-lottieturner) [reply src/psyclone/domain/lfric/lfric_kern.py:618]: is the fault of this PR - DynStencils moved up to be LFRicStencils - fixed [ed1ea8b](https://github.com/stfc/PSyclone/pull/2423/commits/ed1ea8bf5fb9f5d179e129cb62283631656f1e7e)

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 30]
```diff
@@ -0,0 +1,649 @@
...
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors R. W. Ford, A. R. Porter and S. Siso, STFC Daresbury Lab
+# Modified I. Kavcic, A. Coughtrie, L. Turner and O. Brunt, Met Office
+# Modified J. Henrichs, Bureau of Meteorology
+# Modified A. B. G. Chalk and N. Nobre, STFC Daresbury Lab
+
+''' This module implements the PSyclone Dynamo 0.3 API by 1)
```
[the remark this answers is not in the document]
Comment (mo-lottieturner) [reply src/psyclone/domain/lfric/lfric_stencils.py:39]: Yes that looks copied - I knew I had overlooked something!

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 31]
```diff
@@ -0,0 +1,655 @@
...
+from psyclone.configuration import Config
+from psyclone.domain.lfric.lfric_collection import LFRicCollection
+from psyclone.domain.lfric.lfric_constants import LFRicConstants
+from psyclone.errors import GenerationError, InternalError
+from psyclone.f2pygen import (AssignGen, CommentGen, DeclGen,
+                              IfThenGen, TypeDeclGen, UseGen)
+from psyclone.psyir.symbols import ScalarType
+
+
+class LFRicStencils(LFRicCollection):
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:51]: `pycodestyle` reports several errors:
```
src/psyclone/domain/lfric/lfric_stencils.py:321:80: E501 line too long (83 > 79 characters)
src/psyclone/domain/lfric/lfric_stencils.py:331:80: E501 line too long (82 > 79 characters)
src/psyclone/domain/lfric/lfric_stencils.py:377:1: W293 blank line contains whitespace
src/psyclone/domain/lfric/lfric_stencils.py:412:1: W293 blank line contains whitespace
src/psyclone/domain/lfric/lfric_stencils.py:451:1: W293 blank line contains whitespace
```

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 32]
```diff
@@ -0,0 +1,655 @@
...
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors R. W. Ford, A. R. Porter and S. Siso, STFC Daresbury Lab
+# Modified I. Kavcic, A. Coughtrie, L. Turner and O. Brunt, Met Office
+# Modified J. Henrichs, Bureau of Meteorology
+# Modified A. B. G. Chalk and N. Nobre, STFC Daresbury Lab
+
+''' This module implements the stencil information and code generation
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:39]: I would suggest specifying this is for LFRic API. E.g. "... associated with a PSy-layer routine or Kernel stub in LFRic API."

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 33]
```diff
@@ -0,0 +1,655 @@
...
+                dofmap_size_name = self.dofmap_size_symbol(symtab, arg).name
+                parent.add(AssignGen(parent, pointer=True,
+                                     lhs=dofmap_size_name,
+                                     rhs=map_name + "%get_stencil_sizes()"))
+
+    def _declare_maps_invoke(self, parent):
+        '''
+        Declare all stencil maps in the PSy layer.
+
+        :param parent: the node in the f2pygen AST to which to add \
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:531]: Please remove `\` from the description.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 34]
```diff
@@ -0,0 +1,655 @@
...
+                       declarations.
+        :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
+
+        :raises GenerationError: if an unsupported stencil type is encountered.
+        '''
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:536]: Please add a blank space above these triple quotations.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 35]
```diff
@@ -0,0 +1,655 @@
...
+                                   kind=api_config.default_kind["integer"],
+                                   pointer=True,
+                                   entity_decls=[f"{dofmap_size_name}(:) "
+                                                 f"=> null()"]))
+
+    def _declare_maps_stub(self, parent):
+        '''
+        Add declarations for all stencil maps to a kernel stub.
+
+        :param parent: the node in the f2pygen AST representing the kernel \
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:625]: Please remove `\`.

File: src/psyclone/domain/lfric/lfric_stencils.py [hunk 36]
```diff
@@ -0,0 +1,655 @@
...
+                    parent, datatype="integer",
+                    kind=api_config.default_kind["integer"], intent="in",
+                    dimension=",".join([arg.function_space.ndf_name,
+                                        dofmap_size_name]),
+                    entity_decls=[self.dofmap_symbol(symtab, arg).name]))
+
+
+# ---------- Documentation utils -------------------------------------------- #
+# The list of module members that we wish AutoAPI to generate
+# documentation for (See https://psyclone-ref.readthedocs.io).
```
Comment (TeranIvy) [inline src/psyclone/domain/lfric/lfric_stencils.py:654]: Lower case "see", please.

**Review by TeranIvy — CHANGES_REQUESTED** [review]
Almost there, just a few small changes and `pycodestyle` clean-up are required. Please also update the branch with the latest master.
Tests with compilation pass fine.

**Review by TeranIvy — APPROVED** [review]
All well now. Retested with compilation and built examples.

Updated changelog and user guide for merge to master. Integration and compilation tests on PSyclone mirror all pass. Markdown links check started failing again, however this is not the fault of this PR.

Proceeding to merge.

**Comment by TeranIvy** [comment]
Had to push an empty commit to hopefully trigger Coverage report this time.

**Pull request merged**